### PR TITLE
Update light/dark map styles for all map views

### PR DESCRIPTION
**Pull request by jkgz**
- Add light/dark map styles to minimalist styles Avi recently created
- Update map styles when MapViewController loads, not just when traitCollection changes
- Create .day and .night styles using new light/dark map styles so that they can be passed to NavigationViewController in live routing
- Update road label layer logic so that we find the first layer with 'road-label' as a prefix instead of an exact match. 'road-label-simple' can also be used, as in these new styles.
- Set navigationViewController.userNightStyleInDarkMode to true and set it early in lifecycle as per code comment

### CODE REVIEW

File: bikestreets-ios/Map/MapsViewController.swift [hunk 1]
```diff
@@ -130,9 +132,10 @@ class MapsViewController: UIViewController {
   // MARK: -- Layer Placement
   
   internal var belowRoadLabelLayer: LayerPosition? {
-    let id = "road-label"
-    let roadLabelId =  mapView.mapboxMap.style.styleManager.styleLayerExists(forLayerId: id) ? id : nil
-    
+    // Either "road-label" or "road-label-simple" could be used as the road label layer
+    let roadLabelPrefix = "road-label"
+    let roadLabelLayer = mapView.mapboxMap.style.styleManager.getStyleLayers().filter({ $0.id.hasPrefix(roadLabelPrefix) }).first ?? nil
+    let roadLabelId =  roadLabelLayer?.id ?? nil
```
Comment (mattrobmattrob) [inline bikestreets-ios/Map/MapsViewController.swift:138]: Can we use [`first(where:)`](https://developer.apple.com/documentation/swift/array/first(where:)) instead?

```suggestion
    let roadLabelLayer = mapView.mapboxMap.style.styleManager.getStyleLayers().first(where: { $0.id.hasPrefix(roadLabelPrefix) })
    let roadLabelId =  roadLabelLayer?.id
```

File: bikestreets-ios/Styles/BikeStreetsStyles.swift [hunk 2]
```diff
@@ -17,6 +18,43 @@ extension StyleURI {
   static let vamosStreets = StyleURI(
     rawValue: "mapbox://styles/mkbitbucket/clqgwgexa007s01rjhxpwcpaw"
   )!
```
Comment (mattrobmattrob) [inline bikestreets-ios/Styles/BikeStreetsStyles.swift:20]: Can this be deleted now?

File: bikestreets-ios/Styles/BikeStreetsStyles.swift [hunk 3]
```diff
@@ -17,6 +18,43 @@ extension StyleURI {
...
+  
+  static let vamosStreetsDark = StyleURI(
+    rawValue: "mapbox://styles/mkbitbucket/cltt4wy2g01l801oic0udggc7"
+  )!
+}
+
+/// Day style used by NavigationViewController for light mode and day hours.
+/// Will be used in day hours with dark mode enabled unless NavigationViewController.usesNightStyleInDarkMode is true
+/// Additional UIView.appearance modifications can be made in apply(). See CustomStyleUIElements class in Mapbox's Navigation-Examples project for extensive examples.
+class VamosDayStyle: DayStyle {
```
Comment (mattrobmattrob) [inline bikestreets-ios/Styles/BikeStreetsStyles.swift:34]: [nit]
```suggestion
final class VamosDayStyle: DayStyle {
```

File: bikestreets-ios/Styles/BikeStreetsStyles.swift [hunk 4]
```diff
@@ -17,6 +18,43 @@ extension StyleURI {
...
+  }
+  
+  override func apply() {
+    super.apply()
+  }
+}
+
+/// Night style used by NavigationViewController for dark mode, night hours, and tunnels.
+/// Additional UIView.appearance modifications can be made in apply(). See CustomStyleUIElements class in Mapbox's Navigation-Examples project for extensive examples.
+class VamosNightStyle: NightStyle {
```
Comment (mattrobmattrob) [inline bikestreets-ios/Styles/BikeStreetsStyles.swift:48]: [nit]
```suggestion
final class VamosNightStyle: NightStyle {
```

File: bikestreets-ios/Styles/BikeStreetsStyles.swift [hunk 5]
```diff
@@ -17,6 +18,43 @@ extension StyleURI {
...
+class VamosDayStyle: DayStyle {
+  required init() {
+    super.init()
+    mapStyleURL = URL(string: StyleURI.vamosStreetsLight.rawValue)!
+    styleType = .day
+  }
+  
+  override func apply() {
+    super.apply()
+  }
```
Comment (mattrobmattrob) [inline bikestreets-ios/Styles/BikeStreetsStyles.swift:43]: Do we need this since it just proxies the super function?
```suggestion
```

File: bikestreets-ios/Styles/BikeStreetsStyles.swift [hunk 6]
```diff
@@ -17,6 +18,43 @@ extension StyleURI {
...
+class VamosNightStyle: NightStyle {
+  required init() {
+    super.init()
+    mapStyleURL = URL(string: StyleURI.vamosStreetsDark.rawValue)!
+    styleType = .night
+  }
+  
+  override func apply() {
+    super.apply()
+  }
```
Comment (mattrobmattrob) [inline bikestreets-ios/Styles/BikeStreetsStyles.swift:57]: Do we need this since it just proxies the super function?
```suggestion
```

**Review by mattrobmattrob — APPROVED** [review]

File: bikestreets-ios/Styles/BikeStreetsStyles.swift (hunk 2 above)
Comment (jkgz) [reply bikestreets-ios/Styles/BikeStreetsStyles.swift:20]: I suppose so, though I personally would like to see it remain compared to the washed out, hard to parse new style.

File: bikestreets-ios/Styles/BikeStreetsStyles.swift (hunk 6 above)
Comment (jkgz) [reply bikestreets-ios/Styles/BikeStreetsStyles.swift:57]: I left these there because of my comment that more overrides could be done here, but good enough to just add them when needed.

File: bikestreets-ios/Styles/BikeStreetsStyles.swift (hunk 2 above)
Comment (mattrobmattrob) [reply bikestreets-ios/Styles/BikeStreetsStyles.swift:20]: I'm fine leaving it. Perhaps add a comment about the fact that we're testing the other styles?

File: bikestreets-ios/Styles/BikeStreetsStyles.swift (hunk 4 above)
Comment (jkgz) [reply bikestreets-ios/Styles/BikeStreetsStyles.swift:45]: You’ll eventually get me in the habit of using ‘final’. I promise.

File: bikestreets-ios/Map/MapsViewController.swift (hunk 1 above)
Comment (jkgz) [reply bikestreets-ios/Map/MapsViewController.swift:138]: Good call
